### PR TITLE
Hotels: Replace context arrow functions with class functions

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: app/hotels/src/HotelsContext.js [hunk 1]
```diff
@@ -118,19 +118,21 @@ export default { Consumer, Provider };
...
-    };
-
-    // $FlowExpectedError: We need to pass on the navigationOptions if any, flow does not know about it, but a react component might have it
-    if (Component.navigationOptions) {
-      WithHotelsContext.navigationOptions = Component.navigationOptions;
+      render() {
+        return <Consumer>{this.mapStateToProps}</Consumer>;
+      }
     }
+    // $FlowExpectedError: We need to pass on the navigationOptions if any, flow does not know about it, but a react component might have it
```
Comment (RobinCsl) [inline app/hotels/src/HotelsContext.js]: This should probably be removed.

Comment (tbergq) [reply app/hotels/src/HotelsContext.js]: Did you try to remove it? I think it is still needed 😊

Comment (RobinCsl) [reply app/hotels/src/HotelsContext.js]: screen shot 2018-10-31 at 09 00 22
No error it seems.

Comment (tbergq) [reply app/hotels/src/HotelsContext.js]: Oh, sorry, I missed that there was one below as well, this can obviously be removed, good catch.

**Pull request merged**